### DISCUSSION TITLE
输入法能否根据候选项长度自动变换水平模式和竖直模式？

### DISCUSSION THREAD

**Discussion by jkhnfk** [rime/home#1212] [Ideas]
在平时输入过程中有许多时候需要输入较长的内容，如地址、身份证号、税号、以及一些较长的专业术语。
一旦这些内容同时出现在好几个修选项中时，如果此时是水平模式，候选框就会无比地长，观察和选择都比较困难。
能否增加输入法根据候选项的长度，在判断候选项长度比较长时，自动转为竖直模式，待输入完成后又自动转为水平模式？

**Comment by LEOYoon-Tsaw** [comment]
可以，你可以試試，macos上應該不難，windows不知道

**Reply by jkhnfk** [reply]
这可能就是“知识的诅咒”吧。
难者不会，会者不难。我只懂粗浅的Javascript ，研究了许久不得其门。
还望指点一下有何关键点需要去了解掌握。
谢谢。

**Reply by LEOYoon-Tsaw** [reply]
你可以看看代碼，在`SquirrelPanel.m`裏，橫和豎就是一個變量`linear`控制著，根據候選長度動態改就行了（小狼毫我不知道）

**Reply by jkhnfk** [reply]
好的。我先去学习一下。

**Reply by jkhnfk** [reply]
我用的也是squirrel，

**Reply by LEOYoon-Tsaw** [reply]
嗯，在SquirrelPanel下的showPreedit方法裏改theme.linear即可，當然，它可能是只讀的，這就要你動點腦筋了

**Reply by jkhnfk** [reply]
我估计我是搞不来。我只会点Javascript 。
不过还是得感谢您的指点。
